library/systemd/src/modules/systemd_service.rb

```diff
@@ -174,18 +174,118 @@ def restart
 
       # Returns socket associated with service or nil if there is no such socket
       def socket
+        return @socket if @socket
+
         # not triggered
         socket_name = properties.triggered_by
         return unless socket_name
 
         socket_name = socket_name[/\S+\.socket/]
         return unless socket_name # triggered by non-socket
 
-        Yast::SystemdSocket.find(socket_name)
+        @socket = Yast::SystemdSocket.find(socket_name)
+      end
+
+      alias_method :enable_service, :enable
+      private :enable_service
+
+      # Enable a service
+      #
+      # @param mode [Symbol] Start mode (:boot or :demand).
+      # @return [Boolean] true if the service was successfully enabled; false otherwise.
+      def enable
+        self.start_mode = :boot
+      end
+
+      alias_method :disable_service, :disable
+      private :disable_service
+
+      # Disable a service
+      #
+      # If the service has an associated socket, it is disabled too.
+      #
+      # @return [Boolean] true if the service was successfully disabled; false otherwise.
+      def disable
+        self.start_mode = :manual
+      end
+
+      alias_method :enabled_on_boot?, :enabled?
+      private :enabled_on_boot?
+
+      # Determine whether the service is enabled or not
+      #
+      # The service can be enable to be started on boot or on demand.
+      #
+      # @return [Boolean] true if the service is enabled; false otherwise.
+      def enabled?
+        start_mode != :manual
```

Comment:
I don't understand this method. `enabled?` definition is based on `start_mode`, but `start_mode` definition is also based on `enabled?`. Probably I am missing something.

Reply:
No, `start_mode` is not based on `enabled?`. Take into account that `#socket` returns an instance of a different class.

Reply:
Clarified offline. `enabled?` (from parent class) is now `enabled_on_boot?` and `enabled?` has been redefined.

Reply:
this is also backward incompatible change, but I think it is fine.